### PR TITLE
[aspnetcore] Feature/cleanup (contributor rebase)

### PR DESCRIPTION
**Pull request by jimschubert**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

This is a rebase of #2436. Refer to that PR for discussion. This is work toward cleanup defined in #2285.

/cc @A-Joshi for visibility

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java [hunk 1]
```diff
@@ -375,4 +329,69 @@ protected void processOperation(CodegenOperation operation) {
     public String toRegularExpression(String pattern) {
         return escapeText(pattern);
     }
+
+    private void  setCliOption(CliOption cliOption) throws IllegalArgumentException {
+        if (additionalProperties.containsKey(cliOption.getOpt())) {
+            cliOption.setOptValue(additionalProperties.get(cliOption.getOpt()).toString());
+            if (classModifier.getOptValue() == null) {
```
Comment (A-Joshi) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java]: Copy paste error - line should have cliOption.getOptValue not classModifier.getOptvalue

**Comment by A-Joshi** [comment]
Back from my travels, looks good  except for one cut/paste error (I have
commented on the PR). When can t be merged? I have  more changes to
support Async

**Comment by jimschubert** [comment]
Merging master into this branch locally, and regenerating the sample produces invalid annotations. Example:

```
         /// <summary>
         /// User Status
         /// </summary>
         /// <value>User Status</value>
-        [DataMember(Name="userStatus")]
+        [DataMember(Name="userStatus", EmitDefaultValue=)]
         public int? UserStatus { get; set; }
```

@wing328 do you know of a recent change which would have caused this?

I'm also seeing a warning and error dump when running the sample script. Possibly related?

```
$ ./bin/aspnetcore-petstore-server.sh
# START SCRIPT: ./bin/aspnetcore-petstore-server.sh
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=256M; support was removed in 8.0
[main] WARN  i.s.parser.util.DeserializationUtils - Error snake-parsing yaml content
io.swagger.parser.util.DeserializationUtils$SnakeException: Exception safe-checking yaml content  (maxDepth 2000)
	at io.swagger.parser.util.DeserializationUtils$CustomSnakeYamlConstructor.getSingleData(DeserializationUtils.java:300)
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:450)
	at org.yaml.snakeyaml.Yaml.load(Yaml.java:369)
	at io.swagger.parser.util.DeserializationUtils.readYamlTree(DeserializationUtils.java:137)
	at io.swagger.parser.Swagger20Parser.deserializeYaml(Swagger20Parser.java:83)
	at io.swagger.parser.Swagger20Parser.readWithInfo(Swagger20Parser.java:64)
	at io.swagger.parser.SwaggerParser.readWithInfo(SwaggerParser.java:32)
	at io.swagger.v3.parser.converter.SwaggerConverter.readLocation(SwaggerConverter.java:92)
	at io.swagger.parser.OpenAPIParser.readLocation(OpenAPIParser.java:19)
	at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:606)
	at org.openapitools.codegen.cmd.Generate.run(Generate.java:367)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:60)
Caused by: java.lang.IllegalAccessError: tried to access field org.yaml.snakeyaml.constructor.BaseConstructor.composer from class io.swagger.parser.util.DeserializationUtils$CustomSnakeYamlConstructor
	at io.swagger.parser.util.DeserializationUtils$CustomSnakeYamlConstructor.getSingleData(DeserializationUtils.java:279)
	... 11 common frames omitted
[main] WARN  o.o.codegen.utils.ModelUtils - Multiple schemas found in content, returning only the first one
[main] WARN  o.o.codegen.utils.ModelUtils - Multiple schemas found in content, returning only the first one
[main] WARN  o.o.codegen.utils.ModelUtils - Multiple schemas found in content, returning only the first one
```

**Comment by wing328** [comment]
> [main] WARN  i.s.parser.util.DeserializationUtils - Error snake-parsing yaml content

I noticed that too after the parser upgrade.

**Comment by wing328** [comment]
> @wing328 do you know of a recent change which would have caused this?

I also notice that in the latest master and I think I know why. Let me try to fix it later today.

**Comment by wing328** [comment]
> @wing328 do you know of a recent change which would have caused this?

The issue has been fixed via https://github.com/OpenAPITools/openapi-generator/pull/2558

I'll file another PR to decommission the `emitDefaultValue` option as it should be controlled by the nullable attribute instead.

**Comment by wing328** [comment]
FYI. Filed https://github.com/OpenAPITools/openapi-generator/pull/2559 to remove the `emitDefaultValue` option.

**Comment by A-Joshi** [comment]
Looks like it passed all the checks - merge to master hopefully!

**Comment by jimschubert** [comment]
@A-Joshi merged. This should unblock you for any additional work you'd had planned.

**Pull request merged**